src/main/java/com/mongodb/hibernate/internal/translate/mongoast/filter/AbstractAstLogicalFilter.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.internal.translate.mongoast.filter;
+
+import static com.mongodb.hibernate.internal.MongoAssertions.assertFalse;
+
+import java.util.List;
+import org.bson.BsonWriter;
+
+abstract class AbstractAstLogicalFilter implements AstFilter {
```

Comment:
Repeating a comment from a resolved thread:

Let's see if we can stick to a pure-record approach until we find a situation where that doesn't make sense. In this case, the logical operator case is quite similar to the comparison operator case, so we can replace the abstract class and subclasses with a single record and an enum.

This [commit](https://github.com/jyemin/mongo-hibernate/commit/4f0ce43f5a0fbbccee769e92e82fed6f6840a692) shows what I mean.

Reply:
In the future (e.g. in m4), we might need to introduce `$not` in `$expr` context, right? It only needs one filter as constructor parameter, not a collection. We could work around this by
```
new AstLogicalFilter(NOT, List.of(filter));
```
but at least from class design perspective, there is no safety net against `new AstLogicalFilter(NOT, filters)`.
That is why I still prefer the current approach (each kind of logical filter has its own class).

that being said, I have no objection to the proposed approach as well; at least in m3 we won't be bothered with `$not`.

Reply:
committed with very minor changes, including:

- `AstLogicalFilter`'s `filters` non-emptiness assertion
- using `@EnumSource(AstLogicalFilterOperator.class)` in `AstLogicalFilterTests` to simplify a little bit

Reply:
Thanks.

Note though that I used `CsvSource` and not `EnumSource` following the example from `AstComparisonFilterOperationTests`.  My thought was that it was done that way previously in order to more explicitly assert that the correct operator name is used.

Reply:
reverted back to `CsvSource`.

Reply:
> in order to more explicitly assert that the correct operator name is used

One issue issue with using `CsvSource` is that when we add more elements to `AstLogicalFilterOperator`, we have to remember to update `AstLogicalFilterTests.testRendering`, which is not the case with `EnumSource`.

Another issue is that the proposed approach conflates testing two different things in a single unit test. If we want to explicitly test that `AstLogicalFilterOperator.getOperatorName` returns correct values, we should introduce `AstLogicalFilterOperatorTests`, and add tests there for each instance.

Reply:
I think `@CsvSource` originates from `AstComparisonFilterOperationTests` written by me. I made the following changes:

- refactor both `AstLogicalFilterTests` and `AstComparisonFilterOperationTests` to based on `@EnumSource`
- added two operator testing cases (`AstLogicalFilterOperatorTests` and `AstComparisonFilterOperatorTests`); however, in each of them, it is still `@CsvSource` that is used for otherwise we would be testing Java's `Enum`.

Also `AstComparisonFilterOperationTests` was simplified by using existing `FilterTestUtils`.

Reply:
I was trying to avoid re-litigating a decision that we already made in a previous PR, but if this is the direction you want to go, I'm fine with it.  Resolving.